### PR TITLE
Better aspect model

### PR DESCRIPTION
**Pull request by saulvargas**
Please @jacekwasilewski take a look whenever you can at these changes.

### CODE REVIEW

File: RankSys-diversity/src/main/java/es/uam/eps/ir/ranksys/diversity/intentaware/reranking/AlphaXQuAD.java [hunk 1]
```diff
@@ -84,15 +86,16 @@ public UserAlphaXQuAD(Recommendation<U, I> recommendation, int maxLength) {
         protected double nov(Tuple2od<I> iv) {
             return uam.getItemIntents(iv.v1)
                     .mapToDouble(f -> {
-                        return uam.pf_u(f) * uam.pi_f(iv, f) * redundancy.getDouble(f);
-                    }).sum();
+                        return uam.pf_u(f) * iam.pi_f(iv, f) * redundancy.getDouble(f);
```
Comment (jacekwasilewski) [inline RankSys-diversity/src/main/java/es/uam/eps/ir/ranksys/diversity/intentaware/reranking/AlphaXQuAD.java:89]: That is indeed more intuitive - pf_u comes from UserAspectModel and pi_f from ItemAspectModel.

File: RankSys-diversity/src/main/java/es/uam/eps/ir/ranksys/diversity/intentaware/AspectModel.java [hunk 2]
```diff
@@ -89,11 +88,10 @@ public double pf_u(F f) {
     /**
      * Item aspect model for {@link AspectModel}.
      *
-     * @param <U> user type
      * @param <I> item type
      * @param <F> aspect type
      */
-    public interface ItemAspectModel<U, I, F> extends Model<U> {
```
Comment (jacekwasilewski) [inline RankSys-diversity/src/main/java/es/uam/eps/ir/ranksys/diversity/intentaware/AspectModel.java:96]: Doesn't it remove the caching capability?
That would make it slightly inconsistent I think - if caching is used, UAM would be cached and IAM not.
Now I'm thinking that maybe caching could be removed from the Model and if one would like to cache models, could decorate that with e.g. CachedModel.

**Comment by jacekwasilewski** [comment]
Hi @saulvargas 
I see that you already merged the branch - I have added one more comment about the caching.

**Comment by saulvargas** [comment]
The main purpose for this PR was to remove the "cacheability" of the IAM. The IAM is not something of the user exclusively, but of the recommendation as well. Anyway, the constructor of AspectModel assumes no caching as it uses the default constructor of the superclass.

**Pull request merged**